syft/frameworks/torch/tensors/interpreters/large_precision_tensor.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+import torch
+
+from syft.frameworks.torch.overload_torch import overloaded
+from syft.frameworks.torch.tensors.interpreters import AbstractTensor
+
+
+class LargePrecisionTensor(AbstractTensor):
+    """LargePrecisionTensor allows handling of numbers bigger than LongTensor
+
+    Some systems using Syft require larger types than those supported natively. This tensor type supports arbitrarily
+    large values by packing them in smaller ones.
+    If the original tensor is a PyTorch tensor it is not processed
+    """
+
+    def __init__(
+        self, owner=None, id=None, tags=None, description=None, precision=16, virtual_prec=0
+    ):
+        """Initializes a LargePrecisionTensor.
+
+        Args:
+            owner: An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id: An optional string or integer id of the LargePrecisionTensor.
+            precision: The precision this tensor will be transformed to internally. In bits.
+            virtual_prec: The virtual precision required by the caller. In bits.
+        """
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.precision = precision
+        self.virtual_prec = virtual_prec
+
+    @staticmethod
+    def _create_internal_tensor(tensor, precision, virtual_prec):
+        # TODO refs_ok might not be needed as we are passing now torch.tensors here
+        # refs_ok is necessary to enable iterations of reference types.
+        # These big numbers are stored as objects in np
+        result = []
+        for x in np.nditer(tensor, flags=["refs_ok"]):
+            n = int(x.item() * 2 ** virtual_prec)
+            print(f"\nAdding number {n} for item {x.item()}\n")
+            result.append(LargePrecisionTensor._split_number(n, precision))
+        new_shape = tensor.shape + (len(max(result, key=len)),)
+        result = np.array(result).reshape(new_shape)
+        return torch.IntTensor(result)
```

Comment:
should this not be a long tensor for more storage

Reply:
I'll see if by using `dtype` as a param this can be done differently.

Reply:
Using `dtype` now to create the tensor.